File: vaiz/api/projects.py
from vaiz.api.base import BaseAPIClient
from vaiz.models import ProjectsResponse


class ProjectsAPIClient(BaseAPIClient):
    def get_projects(self) -> ProjectsResponse:
        """
        Get all projects in the current space.
        
        Returns:
            ProjectsResponse: The list of projects
        """
        response_data = self._make_request("getProjects", method="POST", json_data={})
        return ProjectsResponse(**response_data) 

File: vaiz-ts-sdk/src/projects.ts
import { BaseAPIClient } from './base';
import { ProjectsResponse, ProjectResponse } from './types';

export class ProjectsAPIClient extends BaseAPIClient {
  
  async getProjects(): Promise<ProjectsResponse> {
    /**
     * Get all projects in the current space.
     * 
     * @returns The list of projects
     */
    const responseData = await this.request<ProjectsResponse>('getProjects', 'POST', {});
    return responseData;
  }

  async getProject(projectId: string): Promise<ProjectResponse> {
    /**
     * Get a single project by its ID.
     * 
     * @param projectId The ID of the project to retrieve
     * @returns The project information
     */
    const responseData = await this.request<ProjectResponse>(
      'getProject', 
      'POST', 
      { projectId }
    );
    return responseData;
  }
}